src/ts/components/subtitleoverlay.ts

```diff
@@ -416,3 +418,81 @@ class ActiveSubtitleManager {
     this.activeSubtitleCueCount = 0;
   }
 }
+
+class SubtitleRegionContainerManager {
+  private subtitleRegionContainers: { [regionName: string]: SubtitleRegionContainer } = {};
+
+  /**
+   * @param subtitleOverlay Reference to the subtitle overlay for adding and removing the containers.
+   */
+  constructor(private subtitleOverlay: SubtitleOverlay) {
+    this.subtitleOverlay = subtitleOverlay;
+  }
+
+  /**
+   * Removes a subtitle label from a container.
+   */
+  removeLabel(labelToRemove: SubtitleLabel): void {
+    for (const regionName in this.subtitleRegionContainers) {
+      this.subtitleRegionContainers[regionName].removeLabel(labelToRemove);
+    }
+  }
+
+  /**
+   * Removes all subtitle containers.
+   */
+  clear(): void {
+    for (const regionName in this.subtitleRegionContainers) {
+      this.subtitleOverlay.removeComponent(this.subtitleRegionContainers[regionName]);
+    }
+
+    this.subtitleRegionContainers = {};
+  }
+
+  /**
+   * Creates and wraps a subtitle label into a container div based on the subtitle region.
+   * If the subtitle has positioning information it is added to the container.
+   * @param label The subtitle label to wrap
+   * @param event Subtitle cue event that contains the positoning information
+   */
+  addLabel(label: SubtitleLabel, event: SubtitleCueEvent): void {
+    const regionName = event.region || 'default';
+    if (!this.subtitleRegionContainers[regionName]) {
+      const regionContainer = new SubtitleRegionContainer({
+        cssClass: `subtitle-position-${regionName}`,
+      });
+
+      this.subtitleRegionContainers[regionName] = regionContainer;
+
+      if (event.regionStyle) {
+        regionContainer.getDomElement().attr('style', event.regionStyle);
+      } else {
+        // getDomElement needs to be called at least once to ensure the component exists
+        regionContainer.getDomElement();
+      }
+
+      for (const regionName in this.subtitleRegionContainers) {
+        this.subtitleOverlay.addComponent(this.subtitleRegionContainers[regionName]);
+      }
+    }
+
+    this.subtitleRegionContainers[regionName].addLabel(label);
+  }
+}
+
+class SubtitleRegionContainer extends Container<ContainerConfig> {
+  constructor(config: ContainerConfig) {
+    const mergedConfig = {...config, cssClasses: [config.cssClass, 'subtitle-region-container']};
```

Comment:
We do have a method for merging configurations: `mergeConfig(...)` (See other components for an example)